### PR TITLE
Remove redundant if and else blocks

### PR DESCRIPTION
**Pull request by Madhu-1**
**What type of PR is this?**

 /kind cleanup

**What this PR does / why we need it**:
This PR removes redundant if and else blocks.

### CODE REVIEW

File: cmd/kube-proxy/app/server.go [hunk 1]
```diff
@@ -233,10 +233,8 @@ func (o *Options) Complete() error {
 		return err
 	}
 
-	if err := utilfeature.DefaultMutableFeatureGate.SetFromMap(o.config.FeatureGates); err != nil {
-		return err
-	}
-	return nil
+	err := utilfeature.DefaultMutableFeatureGate.SetFromMap(o.config.FeatureGates)
+	return err
```
Comment (apelisse) [inline cmd/kube-proxy/app/server.go]: why not `return utilfeature....` everywhere?

Comment (Madhu-1) [reply cmd/kube-proxy/app/server.go]: @apelisse updated PTAL

File: pkg/kubelet/cm/node_container_manager_linux.go [hunk 2]
```diff
@@ -135,10 +135,8 @@ func enforceExistingCgroup(cgroupManager CgroupManager, cName CgroupName, rl v1.
 	if !cgroupManager.Exists(cgroupConfig.Name) {
 		return fmt.Errorf("%q cgroup does not exist", cgroupConfig.Name)
 	}
-	if err := cgroupManager.Update(cgroupConfig); err != nil {
-		return err
-	}
-	return nil
+	err := cgroupManager.Update(cgroupConfig)
```
Comment (apelisse) [inline pkg/kubelet/cm/node_container_manager_linux.go]: missed this one :-)

**Comment by apelisse** [comment]
I found one problem!

Also, how did you do all of these? manually or did you run a script?

File: pkg/kubelet/cm/node_container_manager_linux.go (hunk 2 above)
Comment (Madhu-1) [reply pkg/kubelet/cm/node_container_manager_linux.go]: Thanks for the review. updated  now PTAL

**Comment by Madhu-1** [comment]
> I found one problem!

whats the problem? 
> Also, how did you do all of these? manually or did you run a script?

I did manually.

**Comment by apelisse** [comment]
I was just referring to the `return err` note. Please squash, and I'll lgtm.

**Comment by Madhu-1** [comment]
@apelisse done.

**Comment by apelisse** [comment]
Why would you retest? These are clearly not flakes.

File: pkg/apis/apps/v1/conversion.go [hunk 3]
```diff
@@ -144,10 +144,7 @@ func Convert_v1_RollingUpdateDeployment_To_apps_RollingUpdateDeployment(in *apps
 	if err := s.Convert(in.MaxUnavailable, &out.MaxUnavailable, 0); err != nil {
 		return err
 	}
-	if err := s.Convert(in.MaxSurge, &out.MaxSurge, 0); err != nil {
-		return err
-	}
-	return nil
+	return s.Convert(in.MaxSurge, &out.MaxSurge, 0)
 }
```
Comment (stewart-yu) [inline pkg/apis/apps/v1/conversion.go]: In this file, most of some judgement pattern are copy from auto-generated file, i think it is normal

File: pkg/controller/controller_utils.go [hunk 4]
```diff
@@ -153,11 +153,11 @@ type ControllerExpectations struct {
...
 func (r *ControllerExpectations) GetExpectations(controllerKey string) (*ControlleeExpectations, bool, error) {
-	if exp, exists, err := r.GetByKey(controllerKey); err == nil && exists {
+	exp, exists, err := r.GetByKey(controllerKey)
+	if err == nil && exists {
 		return exp.(*ControlleeExpectations), true, nil
-	} else {
-		return nil, false, err
 	}
+	return nil, false, err
 }
```
Comment (stewart-yu) [inline pkg/controller/controller_utils.go]: ??

**Review by stewart-yu — COMMENTED** [review]
IMO

**Review by thockin — COMMENTED** [review]
Some of these I like and some less so.  Tell me some tool did this for you?

In cases like:

```
  // other code
  if err:= foo(); err != nil {
    return err
  }
  return nil
```

I don't like the change.  I don't intend to say that I want to return the value of foo(), but foo() happens to be the last statement in a series of statements, and if they have all succeeded, return nil.  Subtle difference but important to readers.

Eschewing `else` is generally fine, I guess.

This is a huge PR right before code freeze.  Can we instead a) break this into several smaller PRs that cover one type of transform and b) do it after 1.14 cuts?

**Comment by lavalamp** [comment]
Thank you for your effort on this PR. Some of these changes are clear improvements, but overall I do not think the readability gain is worth the churn.

I would actually go further than Tim-- I'd prefer us not to do mass changes like this without some pressing reason. They are hard to review carefully. I'd prefer a change like this to be preceded by putting some style guide in place? But even if not, I think it's necessary to send this as a series of small changes, not one giant PR. Breaking it up into smaller PRs also means you won't need attention from someone in a high level OWNERS file. :)

**Pull request closed without merging**